### DISCUSSION TITLE
Are there any screenshots about this integration?

### DISCUSSION THREAD

**Discussion by jonasman** [veista/nilan#164] [General]
Hi,

I just bought a new house that comes with a Nilan VP 18 cek (which i think is the same as the VP 18 M2 ek).

Now im trying to decide if i buy the oficial gateway and use the Nilan app OR buy the wifi - modbus bridge (https://www.aliexpress.com/item/4001152539658.html / https://www.waveshare.com/rs485-to-wifi-eth.htm) connect the 3 cables to the 8 pin connector and use this integration in HA to bring all the controls to HomeKit.

Can someone share a few screenshots of how this integration shows up in HA and if you have Homekit, how it shows up in there?

Thanks!

**Comment by veista** [comment]
The available entities depend on the age and model of your Nilan device. Basic climate entity scould be available in all supported devices.

**Reply by jonasman** [reply]
The device is brand new, it comes with the touch screen with software v 3.6.65.32, which i believe is the CTS602
The Heat pump is VP 18 M2 KF EK 3x400V HMI RF (according to its internal plate info). This is in a Finnish house.

Maybe im more curious about:
1. Can i control the air temp via HA / homekit? 
2. Can i control the water temp via HA / homekit?
3. Can i control the water ciculation temp via HA / homekit?
4. Can i monitor all the temperatures shown in the touch screen in HA and maybe Homekit?
5. Can i control the heating mode heating or cooling via HA / homekit?

**Reply by veista** [reply]
Yes to all in HA. I have no experience in the homekit bridge, but to my understanding it should work.

**Reply by jonasman** [reply]
ok thats cool thanks :) 

now i just need to make a decision on what way to go: HA/HK or native Nilan app

I will still wait for a few screenshots from someone to make a final decision

**Reply by veista** [reply]
You can see screenshots in discussions and resolved issues. The problem is that it may not represent your device 100%. From which entities you would like pictures from?

**Reply by jonasman** [reply]
1- air temp
4- monitor some temps
5- heating mode

i will check more resolved issues, i looked into some but didnt find any.

**Reply by jonasman** [reply]
i just bought the wifi bridge and set up the integration. All works well except the floor heating temperature control.
the integration uses a configuration to set the min temperature via `number.nilan_central_heating_min_supply_temperature` , but this cannot be easily exposed to the homekit integration. Any ideas on to expose this as a control just like the HAVC control?

**Reply by veista** [reply]
Hi, you need to study HA templating. You can create a template to do whatever you need.

**Reply by jonasman** [reply]
ok i asked chatGPT on how to build this, but shouldn't the automation do it by ourselves like it does for the hot water tank top and bottom heaters?

another thing i found, my integration thinks my floor heating supports heat pump mode, but mine only has electric or off modes.

**Comment by jonasman** [comment]
just for future reference, here is a screenshot of my Dashboard showing a custom Nilan Card and the HVAC control.

![Screenshot 2025-01-08 at 19 07 24](https://github.com/user-attachments/assets/260011b6-ba10-4d15-af81-6c20dc020938)

**Reply by stauffenberg2020** [reply]
@jonasman ,I have the official Nilan Wifi Gateway and I am using [genvexconnect](https://github.com/superrob/genvexconnect) to connect my Nilan Comfort 300.

Can you please let me know if this Nilan card is present in this integration? or How can I use this Nilan card for my integration?

**Reply by jonasman** [reply]
Hi, the card on the left is custom. So if you have all the values from your integration you can do it too. The one of the right is from the integration.
